src/core/store/modules/tokens/tokens.selectors.ts

```diff
@@ -14,21 +17,22 @@ const selectUserTokensStatusMap = (state: RootState) => state.tokens.statusMap;
 const selectGetTokensStatus = (state: RootState) => state.tokens.statusMap.getTokens;
 const selectGetUserTokensStatus = (state: RootState) => state.tokens.statusMap.user.getUserTokens;
 
-const selectUserTokensMap = (state: RootState) => state.tokens.user.userTokensMap;
-
 /* ----------------------------- Main Selectors ----------------------------- */
-const selectUserTokens = createSelector([selectTokensMap, selectTokensUser], (tokensMap, user): TokenView[] => {
-  const { userTokensAddresses, userTokensMap, userTokensAllowancesMap } = user;
-  const tokens = userTokensAddresses
-    .filter((address) => !!tokensMap[address])
-    .map((address) => {
-      const tokenData = tokensMap[address];
-      const userTokenData = userTokensMap[address];
-      const allowancesMap = userTokensAllowancesMap[address] ?? {};
-      return createToken({ tokenData, userTokenData, allowancesMap });
-    });
-  return tokens.filter((token) => toBN(token.balance).gt(0));
-});
+const selectUserTokens = createSelector(
+  [selectTokensMap, selectTokensUser, selectServicesEnabled],
+  (tokensMap, user, servicesEnabled): TokenView[] => {
+    const { userTokensAddresses, userTokensMap, userTokensAllowancesMap } = user;
+    const tokens = userTokensAddresses
+      .filter((address) => !!tokensMap[address])
+      .map((address) => {
+        const tokenData = tokensMap[address];
+        const userTokenData = userTokensMap[address];
+        const allowancesMap = userTokensAllowancesMap[address] ?? {};
+        return createToken({ tokenData, userTokenData, allowancesMap });
+      });
+    return tokens.filter((token) => toBN(token.balance).gt(0) && (!token.sourceIsZapper || servicesEnabled.zapper));
```

Comment:
Are we tying this too much to zapper?

Reply:
In this case we are only trying to switch zapper on/off, so its intended to be for zapper and not all zaps.

I do think this is not the best place for it, and maybe a common selector would be better, since it mixes two slices of state now... but did not want to overcomplicate the changes.